AR-builder.groovy

```diff
@@ -0,0 +1,456 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+String shellQuote(String s) {
+    // Quote a string so it's suitable to pass to the shell
+    return "'" + s.replaceAll("'", "'\"'\"'") + "'"
+}
+
+def postGithubComment(String changeId, String body) {
+    def authHeader = shellQuote("Authorization: token ${env.GITHUB_API_TOKEN}")
+    def apiUrl = shellQuote("https://api.github.com/repos/wireapp/wire-android-reloaded/issues/${changeId}/comments")
+
+    // The comment body must be quoted for embedding into a JSON string,
+    // and the JSON string must be quoted for embedding into the shell command
+    // line. Note well: the backslash character has a special meaning in
+    // both the first argument (regular expression pattern) and the second
+    // (Matcher.replaceAll() escaping character) of String.replaceAll(); hence,
+    // yet another level of escaping is required here!
+
+    def payload = body.replaceAll('\\\\', '\\\\\\\\').replaceAll('"', '\\\\"').replaceAll('\n', '\\\\n')
+    def json = shellQuote('{"body":"' + payload + '"}')
+
+    // Note the interpolated variables here come from Groovy -- the command
+    // line which the shell interpreter executes is fully rendered, and contains
+    // no unsubstituted variables
+    sh "curl -s -H ${authHeader} -X POST -d ${json} ${apiUrl}"
+
+}
```

Comment:
Maybe **for a next PR**, we can move this to the `Jenkinsfile`.

This way the PR vs branch logic is completely handled in a single file, and we can merge both results in one single PR comment.

This can be achieved by doing something like:

```groovy
// on AR-builder.groovy
uploadToS3(url)
env.APK_URL = url
```

```groovy
// on Jenkinsfile
val result = build(job: 'AR-build-pipeline', [..])
val apkUrl = result.getBuildVariables()["APK_URL"]
```

Then we join all URLs with the build type, and can return a table with results and URLs.